adapters/auth/auth.transformer.ts

```diff
@@ -0,0 +1,29 @@
+import { type User } from '~/interfaces/User'
+import { type SelfInfo } from './auth.type'
+
+export const transformSelfInfoFromApi = (userInfo: any): { info: SelfInfo, user: User } => {
```

Comment:
`userInfo: any` ???

Reply:
We are getting this from API, I don't think we can know the type of userInfo here.

Reply:
add types for now once api is integrated you shall be able to fixes if some types are breaking